src/commands/release/export.ts

```diff
@@ -0,0 +1,110 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { commitOrIdArg } from '.';
+import { Flags } from '@oclif/core';
+import Command from '../../command';
+import * as cf from '../../utils/common-flags';
+import { getBalenaSdk, stripIndent } from '../../utils/lazy';
+import { create } from '@balena/release-bundle';
+import * as fs from 'fs/promises';
+import * as semver from 'balena-semver';
+import { ExpectedError } from '../../errors';
+
+export default class ReleaseExportCmd extends Command {
+	public static description = stripIndent`
+		Saves a release into a file.
+
+        Saving a release to a file allows you to import an exact 
+		copy of the original release into another application.
+
+		Only successful releases can be saved into a file.
+`;
+	public static examples = [
+		'$ balena release export a777f7345fe3d655c1c981aa642e5555 -o ../path/to/release.tar',
+		'$ balena release export 1234567 -o ../path/to/release.tar',
+		'$ balena release export myOrg/myFleet:1.2.3 -o ../path/to/release.tar',
+	];
+
+	public static usage = 'release export <commitOrId>';
+
+	public static flags = {
+		output: Flags.string({
+			description: 'output path',
+			char: 'o',
+			required: true,
+		}),
+		help: cf.help,
+	};
+
+	public static args = {
+		commitOrId: commitOrIdArg({
+			description: 'commit, ID, or version of the release to export',
+			required: true,
+		}),
+	};
+
+	public static authenticated = true;
+
+	public async run() {
+		const { args: params, flags: options } = await this.parse(ReleaseExportCmd);
+
+		const balena = getBalenaSdk();
+
+		let release: balenaSdk.Release;
+		if (
+			typeof params.commitOrId === 'string' &&
+			params.commitOrId.includes(':')
+		) {
+			const fleet = params.commitOrId.split(':')[0];
+			const parsedVersion = semver.parse(params.commitOrId.split(':')[1]);
+			if (parsedVersion == null) {
+				throw new ExpectedError(
+					`Release ${params.commitOrId} could not be exported; version must be valid SemVer.`,
+				);
+			} else {
+				const rawVersion =
+					parsedVersion.build.length === 0
+						? parsedVersion.version
+						: `${parsedVersion.version}+${parsedVersion.build[0]}`;
+				release = await balena.models.release.get(
+					{ application: fleet, rawVersion },
+					{ $select: ['id'] },
+				);
+			}
+		} else {
+			release = await balena.models.release.get(params.commitOrId, {
+				$select: ['id'],
+			});
+		}
+
+		try {
+			const releaseBundle = await create({
+				sdk: balena,
+				releaseId: release.id,
+			});
+			await fs.writeFile(options.output, releaseBundle);
+			console.log(
+				`Release ${params.commitOrId} has been exported to ${options.output}.`,
+			);
+		} catch (error) {
+			throw new ExpectedError(
+				`Release ${params.commitOrId} could not be exported: ${error.message}`,
```

Comment:
Any way to get more explicit than this and share a reason(s) why the release could not be exported?

Reply:
The release bundle module will throw an exception indicating why a release could not be exported.  We have a list of possible reasons here: https://github.com/balena-io-modules/balena-release-bundle/blob/master/src/create.ts#L30-L48